docs/tutorials/excel-tutorial-create-custom-functions.md

```diff
@@ -0,0 +1,203 @@
+
+    1. Open your code editor of choice and navigate to the stock-ticker project folder. 
+    2. Copy and paste the function below and add it to **customfunctions.js**.
+
+    ```js
+    function STOCKPRICE(ticker) {
+        return new Promise(
+            function(resolve) {
+                let xhr = new XMLHttpRequest();
+                let url = "https://api.iextrading.com/1.0/stock/" + ticker + "/price"
+                //add handler for xhr
+                xhr.onreadystatechange = function() {
+                    if (xhr.readyState == XMLHttpRequest.DONE) {
+                    //return result back to Excel
+                    resolve(xhr.responseText);
+                    }
+                }
+                //make request
+                xhr.open('GET', url, true);
+                xhr.send();
+        });
+    }
+    ```
+
+    You'll notice in this code that your asynchronous function returns a JavaScript Promise with the data from the IEX Trading API. Asynchronous custom functions require you to either return a new Promise or use JavaScript's async/await syntax.
+
+    3. In order for Excel to properly run this function, you must add some metadata to the **./config/customfunctions.json** file.
+
+    ```json
+    {
+        "name": "STOCKPRICE",
+        "description": "Multiplies number by 105",
+        "helpUrl": "http://dev.office.com",
+        "result": {
+            "type": "number",
+            "dimensionality": "scalar"
+        },  
+        "parameters": [
+            {
+                "name": "ticker",
+                "description": "stock ticker name",
+                "type": "string",
+                "dimensionality": "scalar"
+            }
+        ],
+    }
+    ```
+    You'll notice that this JSON file describes the function, listing the types and dimensionality of the results and parameters.
+
+    4. You will need to re-register this change once you have saved the file. In Excel, select **Insert > Add-ins > My Add-ins**. This will bring up a list of available add-ins. Under “Developer Add-ins" you will see your add-in, under the name “Excel Custom Function.” Select it to register it.
+
+    In cell B1, run the function `=CONTOSO.STOCKPRICE("MSFT")`. It should show you the stock price for one share of Microsoft stock right now.
+
+## Create a streaming asynchronous custom function
+
+The previous function returned the stock price for Microsoft at a particular moment in time, but stock prices are always changing. With custom functions, it is possible to “stream” data from an API to get updates on stock prices in real time.
+
+To do this, you’ll create a new function, `=CONTOSO.STOCKPRICESTREAM`. It makes a request for updated data every 1000 milliseconds. When a call is made, you may see `#GETTING_DATA` appear in a cell. Once a value is returned, this notification should disappear.
```

Comment:
Here you've put the functions in back ticks. That's fine, but let's be consistent in how we format them.